lua/quickmath/init.lua

```diff
@@ -27,22 +27,47 @@ local function StartSession()
 	vim.api.nvim_buf_attach(0, false, { on_lines = function(...)
 		local content = vim.api.nvim_buf_get_lines(0, 0, -1, true)
 
-		local f, errmsg = loadstring(table.concat(content, "\n"))
-		local success
-		if not f then
-		else
-			success, errmsg = pcall(f)
-		end
+        local function get_result(line, anon)
+            local f, success, errmsg
+
+            if string.find(line, "^anon%d+%s*=") and not anon then
+                errmsg = 'cannot manually assign to anonymous variable'
+            else
+		        f, errmsg = loadstring(line)
+		        if not f then
+		        else
+			        success, errmsg = pcall(f)
+		        end
+            end
+
+            return f, success, errmsg
+        end
 
 		local def = {}
+        local anon_n = 1
 		for i,line in ipairs(content) do
-			if string.find(line, "^%w+%s*=") then
-				local name = string.match(line, "^(%w+)%s*=")
+            local name
+            local anon = false
+            local f, success, errmsg
 
-				table.insert(def, { lnum = i, name = name })
+            if line ~= "" and not string.find(line, "^%-%-.*$") then
```

Comment:
Maybe a misunderstanding. The proposed change was to check for empty lines at the upper level. Otherwise,
empty lines will go to the anonymous values handling clause and will produce the said error.